Restore is failing with MSB3026

Full error:

> [error].packages\microsoft.dotnet.build.tasks.targetframework.sdk\5.0.0-beta.20171.1\build\BinPlace.targets(37,5): error MSB3026: (NETCORE_ENGINEERING_TELEMETRY=Restore) Could not copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.Globalization.Extensions.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.Globalization.Extensions.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.Globalization.Extensions.dll' because it is being used by another process. 

This is happening in both CI and official builds. The error is intermittent though. Seems like an underlying race in restore. 

|Build|Kind|Timeline Record|
|---|---|---|
|[570811](https://dev.azure.com/dnceng/public/_build/results?buildId=570811)|PR https://github.com/dotnet/runtime/pull/33953|Restore|
|[570525](https://dev.azure.com/dnceng/public/_build/results?buildId=570525)|PR https://github.com/dotnet/runtime/pull/33990|Restore|
|[570426](https://dev.azure.com/dnceng/public/_build/results?buildId=570426)|PR https://github.com/dotnet/runtime/pull/33485|Restore|

Evaluated 100 builds
Impacted 3 bulids
Impacted 3 jobs

> runfo search-timeline -d runtime -c 100 -pr -v "MSB3026" -m

So here's a binlog from a failing restore: https://dev.azure.com/dnceng/_apis/resources/Containers/3712946?itemPath=Windows_NT_Libraries%20Build%20Windows_NT%20allConfigurations%20x64%20Debug%2FRestore.binlog

```
Global
    ContinuousIntegrationBuild = True
    TreatWarningsAsErrors = true
    Configuration = Debug
    RepoRoot = F:\workspace\_work\1\s\
    DeployDeps = False
    Build = False
    Rebuild = False
    Test = False
    Pack = False
    IntegrationTest = False
    PerformanceTest = False
    ArchGroup = x64
    TargetArchitecture = x64
    subsetCategory = libraries
    BuildAllConfigurations = true
    RuntimeOS = win10
    RuntimeArtifactsPath = F:\workspace\_work\1\s/artifacts/transport/coreclr
    SkipTestRestore = true
    VersionsPropsPath = F:\workspace\_work\1\s\eng\Versions.props
    __DeployProjectOutput = False
    __BuildPhase = SolutionRestore
    _NETCORE_ENGINEERING_TELEMETRY = Restore
    DefaultBuildAllTarget = Build
    BuildAllProjects = true
    TargetFramework = net472
```
```
Copy
    Assembly = Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
    Parameters
        SourceFiles
            F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                TargetPath = System.IO.Compression.FileSystem.dll
        DestinationFiles
            F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                TargetPath = System.IO.Compression.FileSystem.dll
        SkipUnchangedFiles = True
        OverwriteReadOnlyFiles = True
        UseHardlinksIfPossible = True
    Creating hard link to copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll".
    OutputItems
        FileWrites
            F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                OriginalItemSpec = F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll
                TargetPath = System.IO.Compression.FileSystem.dll

```

```
Global
    ContinuousIntegrationBuild = True
    TreatWarningsAsErrors = true
    Configuration = Debug
    RepoRoot = F:\workspace\_work\1\s\
    DeployDeps = False
    Build = False
    Rebuild = False
    Test = False
    Pack = False
    IntegrationTest = False
    PerformanceTest = False
    ArchGroup = x64
    TargetArchitecture = x64
    subsetCategory = libraries
    BuildAllConfigurations = true
    RuntimeOS = win10
    RuntimeArtifactsPath = F:\workspace\_work\1\s/artifacts/transport/coreclr
    SkipTestRestore = true
    VersionsPropsPath = F:\workspace\_work\1\s\eng\Versions.props
    __DeployProjectOutput = False
    __BuildPhase = SolutionRestore
    _NETCORE_ENGINEERING_TELEMETRY = Restore
    DefaultBuildAllTarget = Build
    BuildAllProjects = true
    TargetFramework = netcoreapp5.0
```

```
Copy
    Assembly = Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
    Parameters
        SourceFiles
            F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                TargetPath = System.IO.Compression.FileSystem.dll
        DestinationFiles
            F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                TargetPath = System.IO.Compression.FileSystem.dll
        SkipUnchangedFiles = True
        OverwriteReadOnlyFiles = True
        UseHardlinksIfPossible = True
    Creating hard link to copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll".
    Could not use a link to copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll". Copying the file instead. Cannot create a file when that file already exists. (0x800700B7)
    Copying file from "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll".
    Errors
        F:\workspace\_work\1\s\.packages\microsoft.dotnet.build.tasks.targetframework.sdk\5.0.0-beta.20171.1\build\BinPlace.targets(37,5): error MSB3026: Could not copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll' because it is being used by another process.  [F:\workspace\_work\1\s\src\libraries\restore\netfx\netfx.depproj]
        F:\workspace\_work\1\s\.packages\microsoft.dotnet.build.tasks.targetframework.sdk\5.0.0-beta.20171.1\build\BinPlace.targets(37,5): error MSB3026: Could not copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll' because it is being used by another process.  [F:\workspace\_work\1\s\src\libraries\restore\netfx\netfx.depproj]
    Creating hard link to copy "F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll" to "F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll".
    OutputItems
        FileWrites
            F:\workspace\_work\1\s\artifacts\bin\ref\net472\System.IO.Compression.FileSystem.dll
                CopyToOutputDirectory = PreserveNewest
                OriginalItemSpec = F:\workspace\_work\1\s\.packages\microsoft.targetingpack.netframework.v4.7.2\1.0.0\lib\net472\System.IO.Compression.FileSystem.dll
                TargetPath = System.IO.Compression.FileSystem.dll
```

So the bug here is that netfx.depproj ends up copying the same file to the same directory when building for `net472` and `netcoreapp5.0`.

--- a/src/libraries/restore/netfx/netfx.depproj
+++ b/src/libraries/restore/netfx/netfx.depproj
@@ -8,15 +8,13 @@
     <NETStandardSupportPackageVersion>2.0.1-servicing-26011-01</NETStandardSupportPackageVersion>
     <NETStandardSupportRoot>$(NuGetPackageRoot)$(NETStandardSupportPackageId.ToLower())\$(NETStandardSupportPackageVersion)\build</NETStandardSupportRoot>
     <AddNetStandardSupportPackage Condition="'$(TargetFramework)' == 'net461' or '$(TargetFramework)' == 'net462' or '$(TargetFramework)' == 'net47'">true</AddNetStandardSupportPackage>
-    <TargetFrameworks>net45;net451;net46;net461;net462;net47;net471;net472;$(netcoreappCurrent)</TargetFrameworks>
+    <TargetFrameworks>net45;net451;net46;net461;net462;net47;net471;net472</TargetFrameworks>
+    <!-- if we aren't building all configurations, ensure we populate NetFxRefPath for netcoreapp by building this project -->
+    <TargetFrameworks Condition="'$(BuildAllConfigurations)' != 'true'">$(TargetFrameworks);$(NetCoreAppCurrent)</TargetFrameworks>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetFramework)' != ''">
-    <TargetFrameworks Condition="$(TargetFramework.Contains('net4'))">net45;net451;net46;net461;net462;net47;net471;net472</TargetFrameworks>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetsNetFx)' != 'true'">
+  <PropertyGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
     <!-- For things not .NETFramework we need the net472 targeting pack to generate facades -->
     <AssetTargetFallback>$(NetFrameworkCurrent)</AssetTargetFallback>
-    <NoWarn>$(NoWarn);NU1701</NoWarn>
     <TargetingPackNugetPackageId>Microsoft.TargetingPack.NETFramework.v4.7.2</TargetingPackNugetPackageId>
   </PropertyGroup>
 
@@ -67,7 +65,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="$(TargetingPackNugetPackageId)">
+    <PackageReference Include="$(TargetingPackNugetPackageId)" NoWarn="NU1701">
       <Version >$(_TargetingPackVersion)</Version>
     </PackageReference>
     <PackageReference Include="System.Memory" Condition="'$(TargetFramework)' == 'net45' or '$(TargetFramework)' == 'net46'">

